doc/source/whatsnew/v0.24.2.rst

```diff
@@ -24,6 +24,8 @@ Fixed Regressions
 
 - Fixed issue in ``DataFrame`` construction with passing a mixed list of mixed types could segfault. (:issue:`25075`)
 
+- Fixed regression in :meth:`DataFrame.apply` causing RecursionError when classes like ``dict`` were passed as argument. (:issue:`25196`)
```

Comment:
Wording nits for clarity

```suggestion
- Fixed regression in :meth:`DataFrame.apply` causing RecursionError when ``dict``-like class were passed as argument. (:issue:`25196`)
```

Reply:
it's not a "dict-like class" causing the issue, it's really `dict` as a class name, or `Series` or `defaultdict` and so on

Reply:
Two other small things:
 - RecursionError --> \`\`RecursionError\`\`
 - Can you delete the blank lines between entries (lines 24/26), as blank lines aren't required between entries.

Reply:
Hmm not sure I follow; the distinction you are trying to make is between passing a type and an instance as an argument, right?